x/storage/keeper/payment.go

```diff
@@ -366,6 +426,19 @@ func (k Keeper) ChargeViaObjectChange(ctx sdk.Context, bucketInfo *storagetypes.
 
 	userFlows.Flows = append(userFlows.Flows, getNegFlows(preOutFlows)...)
 	userFlows.Flows = append(userFlows.Flows, newOutFlows...)
+
+	if ctx.IsUpgraded(upgradetypes.Pawnee) {
```

Comment:
Most buckets do not have rate limit. Can we check if there is rate limiter earlier and break quickly? Otherwise, it seems it will affect the performance for the general cases.

Reply:
there is no heavy operations under this function, i think it's fine here